### PR TITLE
Fix/packit service/1328 bis

### PR DESCRIPTION
**Pull request by majamassarini**
Fixes https://github.com/packit/packit-service/issues/1328 (point 3)

The new db diagram:
ERDiagram

### CODE REVIEW

File: packit_service/models.py [hunk 1]
```diff
@@ -1546,27 +1562,56 @@ def get_all_by(
...
+            .join(
+                PipelineModel,
+                PipelineModel.copr_build_group_id == CoprBuildGroupModel.id,
+            )
+            .join(
+                ProjectEventModel,
+                PipelineModel.project_event_id == ProjectEventModel.id,
+            )
+            .filter(CoprBuildTargetModel.project_name == project_name)
+            .filter(ProjectEventModel.commit_sha == commit_sha)
```
Comment (majamassarini) [inline packit_service/models.py:1580]: Are we ok with having lower performance here? (And likely in every other build/test model class)

File: packit_service/models.py [hunk 2]
```diff
@@ -915,7 +956,7 @@ class PullRequestModel(BuildsAndTestsConnector, Base):
     project = relationship("GitProjectModel", back_populates="pull_requests")
 
     job_config_trigger_type = JobConfigTriggerType.pull_request
-    job_trigger_model_type = JobTriggerModelType.pull_request
+    project_event_model_type = ProjectEventModelType.pull_request
 
     @classmethod
     def get_or_create(
```
Comment (majamassarini) [inline packit_service/models.py:962]: Do we want to pass a `commit_sha` here? (And also when creating a IssueModel, GitBranchModel and ProjectReleaseModel?)

File: packit_service/models.py [hunk 3]
```diff
@@ -1201,20 +1249,25 @@ class PipelineModel(Base):
     sync_release_run = relationship("SyncReleaseModel", back_populates="runs")
 
     @classmethod
-    def create(cls, type: JobTriggerModelType, trigger_id: int) -> "PipelineModel":
+    def create(
+        cls, type: ProjectEventModelType, event_id: int, commit_sha: str
+    ) -> "PipelineModel":
```
Comment (majamassarini) [inline packit_service/models.py:1254]: Is it ok to pass a `commit_sha` when creating  a new `PipelineModel`?

**Comment by majamassarini** [comment]
recheck

**Comment by jpopelka** [comment]
recheck

**Comment by majamassarini** [comment]
recheck

**Comment by majamassarini** [comment]
recheck

**Comment by majamassarini** [comment]
recheck

**Comment by majamassarini** [comment]
recheck

File: packit_service/models.py [hunk 4]
```diff
@@ -1136,10 +1163,13 @@ def __repr__(self):
 
 class ProjectEventModel(Base):
     """
-    Model representing a project event which triggers some packit task.
+    Model representing a "project event" which triggers some packit task.
+    Like a push into a pull request: the push is a "project event" with a
+    given commit sha into a specific "project object" wich is a pull request.
```
Comment (lbarcziova) [inline packit_service/models.py:1168]:
```suggestion
    Like a push into a pull request: the push is a "project event" with a
    given commit sha into a specific "project object" which is a pull request.
```

File: alembic/versions/af00a7f19025_move_commit_sha_to_project_events.py [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+
+    op.drop_index("ix_copr_build_targets_commit_sha", table_name="copr_build_targets")
+    op.drop_column("copr_build_targets", "commit_sha")
+    op.drop_column("koji_build_targets", "commit_sha")
+    op.drop_column("srpm_builds", "commit_sha")
+    op.drop_column("tft_test_run_targets", "commit_sha")
+    op.drop_index(
+        "ix_vm_image_build_targets_commit_sha", table_name="vm_image_build_targets"
+    )
+    op.drop_column("vm_image_build_targets", "commit_sha")
```
Comment (lbarcziova) [inline alembic/versions/af00a7f19025_move_commit_sha_to_project_events.py:231]: nice that we could move this into one table! :+1:

File: alembic/versions/af00a7f19025_move_commit_sha_to_project_events.py [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+    )
+    op.create_index(
+        "ix_vm_image_build_targets_commit_sha",
+        "vm_image_build_targets",
+        ["commit_sha"],
+        unique=False,
+    )
+    op.add_column(
+        "tft_test_run_targets",
+        sa.Column("commit_sha", sa.VARCHAR(), autoincrement=False, nullable=True),
```
Comment (lbarcziova) [inline alembic/versions/af00a7f19025_move_commit_sha_to_project_events.py:249]:
```suggestion
        sa.Column("commit_sha", sa.String(), autoincrement=False, nullable=True),
```
weren't these Strings previously?

File: packit_service/models.py [hunk 7]
```diff
@@ -356,14 +365,24 @@ class ProjectAndTriggersConnector:
...
         return project_event.get_project_event_object() if project_event else None
 
     def get_project(self) -> Optional["GitProjectModel"]:
         project_event_object = self.get_project_event_object()
         return project_event_object.project if project_event_object else None
 
+    @property
+    def commit_sha(self) -> Optional[str]:
+        project_event_model = self.get_project_event_model()
+        return project_event_model.commit_sha
```
Comment (lbarcziova) [inline packit_service/models.py:379]: could `project_event_model` here be None? (and the same for the setter)

**Review by lbarcziova — COMMENTED** [review]
this is admirable work Maja!! :rocket:

just a few notes, it would be nice to get this to staging and test it there

as for the migration, have you tested that with the production data?

**Comment by majamassarini** [comment]
> as for the migration, have you tested that with the production data?

Yes, on my laptop with the production data the migration took more than half an hour but less than an hour.

**Comment by majamassarini** [comment]
recheck

**Review by lbarcziova — APPROVED** [review]
let's ship it! 

thanks Maja for taking this complext issue!

**Pull request merged**